components/bio-formats-plugins/src/loci/plugins/out/Exporter.java

```diff
@@ -546,12 +552,21 @@ else if (cal.frameInterval != 0.0 && cal.getTimeUnit().equals("sec")) {
                 String base = outfile.substring(0, dot);
                 String ext = outfile.substring(dot);
 
+                String zPlaces = "%d";
+                String tPlaces = "%d";
+                String cPlaces = "%d";
+                if (padded) {
+                  zPlaces = "%0" + String.valueOf(sizeZ).length() + "d";
+                  tPlaces = "%0" + String.valueOf(sizeT).length() + "d";
+                  cPlaces = "%0" + String.valueOf(sizeC).length() + "d";
+                }
+                
                 int nextFile = 0;
                 for (int z=0; z<(splitZ ? sizeZ : 1); z++) {
                     for (int c=0; c<(splitC ? sizeC : 1); c++) {
                         for (int t=0; t<(splitT ? sizeT : 1); t++) {
-                            outputFiles[nextFile++] = base + (splitZ ? "_Z" + z : "") +
-                                    (splitC ? "_C" + c : "") + (splitT ? "_T" + t : "") + ext;
+                            outputFiles[nextFile++] = base + (splitZ ? "_Z" + String.format(zPlaces, z) : "") +
```

Comment:
What is preventing us from consuming  `FormatTools.getFilename()` directly here: is it because it requires an `IFormatReader` that we don't have at this point?